src/util/codec/datum.rs

```diff
@@ -47,11 +48,62 @@ pub enum Datum {
     Max,
 }
 
+impl Display for Datum {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match *self {
+            Datum::Null => write!(f, "NULL"),
+            Datum::I64(i) => write!(f, "I64({})", i),
+            Datum::U64(u) => write!(f, "U64({})", u),
+            Datum::F64(v) => write!(f, "F64({})", v),
+            Datum::Dur(ref d) => write!(f, "Dur({})", d),
+            Datum::Bytes(ref bs) => write!(f, "Bytes(\"{}\")", escape(bs)),
+            Datum::Dec(ref d) => write!(f, "Dec({})", d),
+            Datum::Min => write!(f, "MIN"),
+            Datum::Max => write!(f, "MAX"),
+        }
+    }
+}
+
+impl Debug for Datum {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "{}", self)
+    }
+}
+
 fn cmp_f64(l: f64, r: f64) -> Result<Ordering> {
     l.partial_cmp(&r)
         .ok_or_else(|| invalid_type!("{} and {} can't be compared", l, r))
 }
 
+#[cfg(debug_assertions)]
+macro_rules! abs {
+    ($r:ident) => {
+        // in debug mode, if r is `i64::min_value()`, `-r` will panic.
+        // but in release mode, `-r as u64` will get `|r|`.
+        if $r == i64::min_value() {
+            i64::max_value() as u64 + 1
+        } else {
+            -$r as u64
+        }
+    };
+}
+
+#[cfg(not(debug_assertions))]
+macro_rules! abs {
+    ($r:ident) => {
+        -$r as u64
```

Comment:
Because it's not for general purpose usage. It's just an workaround for the situation described at L81. And it's only used at L103.